packages/patternfly-react/src/components/VerticalNav/VerticalNav.js

```diff
@@ -334,12 +333,10 @@ class BaseVerticalNav extends React.Component {
         </VerticalNavItem>
       ));
     const itemComponents = itemsFromProps || itemsFromChildren || [];
-
     const masthead = findChild(
       children,
-      child => child.type === VerticalNavMasthead
+      child => child.type.name === 'Masthead'
```

Comment:
This is going to make this a breaking change.  Also, I am pretty sure this will not work in production code.  React/Uglify strip out a lot of this.  Comparing by name is never really a good idea. Since this is breaking, I would rather see masthead be a prop that gets passed instead of searched for in children
```
<VerticalNav
   Masthead={<Masthead />}
/>
```
This is more explicit, saves on iterating children every render, removes the need for a `hideMasthead` prop, and allows users to render any Masthead.